packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -195,15 +195,33 @@ class IOSDevice extends Device {
     }
 
     // Step 3: Attempt to install the application on the device.
-    int installationResult = await runCommandAndStreamOutput(<String>[
+    List<String> launchArguments = <String>[];
+
+    if (debuggingOptions.startPaused)
+      launchArguments.add("--start-paused");
+
+    if (debuggingOptions.debuggingEnabled) {
+      launchArguments.add("--enable-checked-mode");
+      int port = await debuggingOptions.findBestObservatoryPort();
```

Comment:
I think we'll need to have the device bind to a port, have it print out the port number, use the logging to scrape the port number, bind to a local port (`findBestObservatoryPort()`), and set up port forwarding from the host to the device. There's a bunch of logic to do this already for android, it'll all be reusable.